src/org/daisy/dotify/formatter/impl/writer/PEFMediaWriter.java

```diff
@@ -227,9 +243,30 @@ public void newRow(Row row) {
             errorCount++;
         }
         pst.print("<row");
+
+        Object externalReference = row.getExternalReference();
+        if (externalReference instanceof Map) {
+            Map<QName, String> metadata = (Map<QName, String>) externalReference;
+            if (metadata != null && !metadata.isEmpty()) {
+                for (Map.Entry<QName, String> entry : metadata.entrySet()) {
+                    QName name = entry.getKey();
+
+                    String prefix = namespaces.get(name.getNamespaceURI());
+                    if (prefix == null) {
+                        prefix = "ns" + namespaceCounter++;
+                        pst.print(" xmlns:" + prefix + "=\"" + name.getNamespaceURI() + "\"");
+                    }
+
+                    pst.print(" " + prefix + ":" + name.getLocalPart() + "=\"" + entry.getValue() + "\"");
```

Comment:
Looks good. With a bit of reorganization wouldn't it have been possible to reuse the `getNamespaces()` method above though? Haven't looked at it closely.

Reply:
These namespaces for the global state is already removed when that function is executed. It gets namespaces for the meta lines in the header.

Reply:
I know.